### ISSUE TITLE
UseAzureSignalR should implement /negotiate end point 

### ISSUE THREAD

**Issue by davidfowl** [Azure/azure-signalr#14]
To make the integration as seamless as possible, we want the SignalR client to connect to the application like normal, but then get a negotiate response that redirects them to the service URL. UseAzureSignalR should implement minimal negotiate endpoint that does this.

The other complication is the direct websocket connection scenario. Today if you specify websockets on the client side, it will skip negotiation and directly connect via websockets to the service endpoint. To mitigate this, we should handle incoming websocket connections with the SignalR client and send a redirect response to the service URL (with the access token built into the URL).

Here's the link to the issue to update the client logic on the SignalR side https://github.com/aspnet/SignalR/issues/1947.

**Comment by mhintzke** [comment]
@davidfowl I know this is very old, but we ran into an issue today that we think is a bug, but I am starting to get second thoughts. Our application code is using `Microsoft.AspNetCore.Builder.AzureSignalRApplicationBuilderExtensions.UseAzureSignalR`, but when we did that I ran into #1602. I reproduced that "bug" in a forked repository but while I was doing it I realized that the sample application doesn't even use `UseAzureSignalR` at all. I hand-waved it and continued reproducing the bug which was a success, however, after coming back and trying to use `UseEndpoints` instead to do the exact same thing all of a sudden everything worked.

This makes sense why it works (because now the Authorization system can successfully see the `AuthorizeAttribute` and add to endpoint metadata), but what is the purpose of `UseAzureSignalR` if it appears to lack a significant amount of logic and has Authorization issues (among more I assume).

I think using `UseEndpoints` will be the solution to our problem, but I wonder if it is time to deprecate `UseAzureSignalR` so others to fall into the same trap as myself.

Thoughts?
